test/tests/functional/pbs_ralter.py

```diff
@@ -2753,3 +2753,45 @@ def test_alter_force_select(self):
         self.alter_a_reservation(rid, start, end, select="1:ncpus=1",
                                  a_duration=20, extend='force',
                                  whichMessage=0)
+
+    def test_ralter_force_start_end_running_resv(self):
+        """
+        Test that forcefully altering a running reservation takes effect.
+        """
+
+        duration = 3600
+        offset = 20
+
+        rid, start, end = self.submit_and_confirm_reservation(
+            offset, duration, select="2:ncpus=4")
+
+        resv_queue = rid.split('.')[0]
+        a = {'queue': resv_queue}
+        j = Job(attrs=a)
+        jid = self.server.submit(j)
+
+        self.logger.info('Waiting for reservation to start')
+        a = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')}
+        off = int(start - time.time())
+        self.server.expect(RESV, a, id=rid, offset=off)
+
+        self.alter_a_reservation(rid, start, end, confirm=False, shift=-10,
+                                 alter_s=True, extend='force', whichMessage=0)
+
+        self.alter_a_reservation(rid, start, end, confirm=False, shift=-100,
+                                 alter_e=True, extend='force')
+        _, _, t_end = self.get_resv_time_info(rid)
+        end = end - 100
+        self.assertEqual(int(t_end), end)
+
+        self.alter_a_reservation(rid, start, end, confirm=False,
+                                 a_duration=4000, extend='force', sequence=2)
```

Comment:
Usually you have to bump sequence for each alter you do.  This is the 3rd alter, why didn't you need  to bump it in the last call?

Reply:
Actually first alter command is rejected because there is a running job inside this reservation. This is the reason the sequence number here is 2.

Reply:
I've added a comment to make it clear.